### PR TITLE
Payment Entry

### PR DESCRIPTION
**Pull request by nabinhait**

### CODE REVIEW

**Comment by nabinhait** [comment]
@rmehta test cases passed

File: erpnext/accounts/doctype/payment_entry/payment_entry.js [hunk 1]
```diff
@@ -0,0 +1,724 @@
...
+	},
+	
+	refresh: function(frm) {
+		frm.events.hide_unhide_fields(frm);
+		frm.events.set_dynamic_labels(frm);
+		frm.events.show_general_ledger(frm);
+	},
+	
+	hide_unhide_fields: function(frm) {
+		var company_currency = frappe.get_doc(":Company", frm.doc.company).default_currency;
```
Comment (rmehta) [inline erpnext/accounts/doctype/payment_entry/payment_entry.js]: depends_on ?

File: erpnext/accounts/doctype/payment_entry/payment_entry.py [hunk 2]
```diff
@@ -0,0 +1,666 @@
...
+		for d in self.get("references"):
+			if d.allocated_amount and d.reference_doctype == "Journal Entry":				
+				je_accounts = frappe.db.sql("""select debit, credit from `tabJournal Entry Account`
+					where account = %s and party=%s and docstatus = 1 and parent = %s
+					and (reference_type is null or reference_type in ("", "Sales Order", "Purchase Order"))
+					""", (self.party_account, self.party, d.reference_name), as_dict=True)
+
+				if not je_accounts:
+					frappe.throw(_("Row #{0}: Journal Entry {0} does not have account {1} or already matched against another voucher")
+						.format(d.idx, d.reference_name, self.party_account))
```
Comment (rmehta) [inline erpnext/accounts/doctype/payment_entry/payment_entry.py]: formatting index error

File: erpnext/accounts/doctype/payment_entry/payment_entry.py [hunk 3]
```diff
@@ -0,0 +1,666 @@
...
+	
+	return {
+		"party_account": party_account,
+		"party_account_currency": account_currency,
+		"party_balance": party_balance,
+		"account_balance": account_balance
+	}
+
+@frappe.whitelist()	
+def get_account_details(account, date):
```
Comment (rmehta) [inline erpnext/accounts/doctype/payment_entry/payment_entry.py]: `frappe.has_permission('Payment Entry', throw=True)`

**Comment by cgpurbaugh** [comment]
What is the purpose of this doctype?  Are checks to be made from it?  If so, can 'in words' be added?

**Pull request merged**